sirepo_detector.py

```diff
@@ -194,28 +209,46 @@ def connect(self, sim_id):
         else:
             field1 = None
 
-        # First define a factory
-        optic_id = self.find_optic_id_by_name(optic_name, data)
-        schema = {f'sirepo_{k}': v for k, v in
-                  data['models']['beamline'][optic_id].items()}
-
         def class_factory(cls_name):
-            dd = {k: Cpt(SynAxis) for k in schema}
+            dd = {k: Cpt(SynAxis) for k in self.parameters}
             return type(cls_name, (Device,), dd)
 
-        SirepoComponent = class_factory('SirepoComponent')
-        sirepo_component = SirepoComponent(name=optic_name)
+        sirepo_components = {}
 
-        self.sirepo_component = sirepo_component
+        # Create sirepo component for each optical element, set active element
+        # to the one selected by the user
+        for i in range(len(data['models']['beamline'])):
```

Comment:
This can be extended even more to include different reports, like `Single-Electron Spectrum Report, ***m` (the corresponding name in Sirepo's schema is `intensityReport`)